rotkehlchen/exchanges/woo.py

```diff
@@ -0,0 +1,537 @@
+        )
+        return movements
+
+    def validate_api_key(self) -> tuple[bool, str]:
+        """Validates that the Woo API key is good for usage in rotki"""
+        try:
+            response = self._api_query('v1/client/trades', options={'limit': 1})
+        except RemoteError as e:
+            return False, str(e)
+        if response.status_code != HTTPStatus.OK:
+            return self._process_unsuccessful_response(
+                response=response,
+                case='validate_api_key',
+            )
+        return True, ''
+
+    def _api_query(
+            self,
+            endpoint: str,
+            method: Literal['GET', 'POST'] = 'GET',
+            options: Optional[dict[str, Any]] = None,
+    ) -> Response:
+        """
+        Request a  Woo API endpoint (from `endpoint`).
+
+        May Raise:
+        - RemoteError
+        """
+        call_options = options if options else {}
+        request_url = f'{self.base_uri}/{endpoint}'
+        timestamp = str(ts_now_in_ms())
+        parameters = urllib.parse.urlencode(call_options)
+        normalized_content = f'{timestamp}{method}/{endpoint}{parameters}' if endpoint.startswith('v3') else f'{parameters}|{timestamp}'  # noqa: E501
+        signature = hmac.new(
+            self.secret,
+            msg=normalized_content.encode('utf-8'),
+            digestmod=hashlib.sha256,
+        ).hexdigest()
+        self.session.headers.update({
+            'x-api-signature': signature,
+            'x-api-timestamp': timestamp,
+        })
+        log.debug('Woo API request', request_url=request_url, options=options)
+        try:
+            response = self.session.request(
+                method=method,
+                url=request_url,
+                data=call_options if method == 'POST' else {},
+                params=call_options if method == 'GET' else {},
+                headers=self.session.headers,
+            )
+        except requests.exceptions.RequestException as e:
+            raise RemoteError(
+                f'Woo {method} request at {request_url} connection error: {e}.',
+            ) from e
+
+        return response
+
+    @overload
+    def _api_query_paginated(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+            case: Literal['trades'],
+    ) -> list[Trade]:
+        ...
+
+    @overload
+    def _api_query_paginated(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+            case: Literal['asset_movements'],
+    ) -> list[AssetMovement]:
+        ...
+
+    def _api_query_paginated(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+            case: Literal['trades', 'asset_movements'],
+    ) -> Union[list[Trade], list[AssetMovement], list]:
+        """Request a Woo API endpoint paginating via an options attribute."""
+        case_pretty = case.replace('_', ' ')  # human readable case
```

Comment:
Why do this? Why not just pass `trades` and `asset movements` as the case strings? Remove this "pretty thing"